hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/SecurityUtil.java

```diff
@@ -586,17 +594,62 @@ InetAddress getByName(String hostname) throws UnknownHostException {
       return hostResolver.getByName(hostname);
     }
   }
-  
+
   interface HostResolver {
-    InetAddress getByName(String host) throws UnknownHostException;    
+    InetAddress getByName(String host) throws UnknownHostException;
+  }
+
+  static abstract class CacheableHostResolver implements HostResolver {
+    private volatile LoadingCache<String, InetAddress> cache;
+
+    CacheableHostResolver(long expiryIntervalSecs) {
+      if (expiryIntervalSecs > 0) {
+        cache = CacheBuilder.newBuilder()
+            .expireAfterWrite(expiryIntervalSecs, TimeUnit.SECONDS)
```

Comment:
For the cache, should we set a configuration for the maximum number of entries?

Reply:
Thanks for the suggestion!

Since the number of nodes in the cluster is limited, the cache size will not grow unbounded in practice. To keep the configuration simple, I think it’s reasonable not to introduce an additional setting for the maximum number of entries at this point.

Also, note that the cache in NodesListManager.CachedResolver doesn’t enforce a size limit either, which seems to work fine under similar assumptions.

Reply:
Thanks for the explanation! this makes sense to me.